### PR TITLE
Make interface versions configurable

### PR DESCRIPTION
**Pull request by Deedone**
Add support for configuring xen, sysctl, and domctl interface versions.

### CODE REVIEW

File: include/zephyr/xen/public/grant_table.h [hunk 1]
```diff
@@ -126,7 +126,7 @@ typedef uint32_t grant_ref_t;
  * Version 1 of the grant table entry structure is maintained purely
  * for backwards compatibility.  New guests should use version 2.
  */
-#if __XEN_INTERFACE_VERSION__ < 0x0003020a
+#if CONFIG_XEN_INTERFACE_VERSION < 0x0003020a
```
Comment (firscity) [inline include/zephyr/xen/public/grant_table.h:129]: Maybe it is easier to add `#define __XEN_INTERFACE_VERSION__ CONFIG_XEN_INTERFACE_VERSION` somewhere?

File: arch/arm64/core/xen/Kconfig [hunk 2]
```diff
@@ -25,3 +25,11 @@ config XEN_DOM0LESS
 	help
 	  Configures Zephyr as DomU, that can be started on Dom0less
 	  setup.
+
+config XEN_INTERFACE_VERSION
+	hex "Xen interface version"
+	default 0x00040e00
```
Comment (firscity) [inline arch/arm64/core/xen/Kconfig:31]: I'm not sure that we need to add default value, but for sure not 4.13 (all changes in mainline and here were targeted to 4.17)

File: include/zephyr/xen/public/xen.h [hunk 3]
```diff
@@ -41,6 +41,11 @@ DEFINE_XEN_GUEST_HANDLE(char);
 __DEFINE_XEN_GUEST_HANDLE(uchar, unsigned char);
 DEFINE_XEN_GUEST_HANDLE(int);
 __DEFINE_XEN_GUEST_HANDLE(uint,  unsigned int);
+
+#ifndef __XEN_INTERFACE_VERSION__
```
Comment (firscity) [inline include/zephyr/xen/public/xen.h]: Is it actually should be in `#ifndef __ASSEMBLY__` block?

File: include/zephyr/xen/public/grant_table.h (hunk 1 above)
Comment (GrygiriiS) [reply include/zephyr/xen/public/grant_table.h:129]: It seems, the same approach for XEN_DOMCTL_INTERFACE_VERSION also could be right thing to do?

**Comment by GrygiriiS** [comment]
`Reviewed-by: Grygorii Strashko <grygorii_strashko@epam.com>`

And you probably want to update PR description.

**Review by firscity — APPROVED** [review]
`Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>`

Also, first patch for sure should be upstreamed

**Pull request merged**